src/main/kotlin/calculator/Parser.kt

```diff
@@ -0,0 +1,30 @@
+package calculator
+
+object Parser {
+    private val defaultDelimiters: List<String> = listOf(",", ":")
+    private var delimiters: List<String> = defaultDelimiters
+
+    private fun addDelimiters(vararg delimiters: String) {
+        this.delimiters = this.delimiters.plus(delimiters)
+    }
+
+    private fun extractCustomDelimitersIfExist(value: String): String {
+        if (!value.startsWith("//")) return value
+        val customSeparator = value.substringAfter("//").substringBefore("\\n")
+        if (customSeparator == ".") throw IllegalArgumentException()
+        addDelimiters(customSeparator)
+        return value.substringAfter("\\n")
+    }
+
+    fun parseToNumberList(value: String): List<Double> {
+        val valueAfterExtractCustomDelimiters = extractCustomDelimitersIfExist(value)
+        val splitList = valueAfterExtractCustomDelimiters.split(*delimiters.toTypedArray())
```

Comment:
```suggestion
    fun parseToNumberList(value: String): List<Double> {
        val numberString = extractNumbers(value)
        val numbers: List<String> = numberString.split(*delimiters.toTypedArray())
```

변수 또한 `결과값` 에만 집중해주세요!

Reply:
변수에 과정이 아닌 결과를 담음으로서 훨씬 명료해지겠네요. 반영했습니다!
17b97837eb93ec4462b9ac34663a5e3b25995f66